app/grandchallenge/components/views.py

```diff
@@ -153,3 +156,33 @@ def get_success_message(self, cleaned_data):
             success_message=self.success_message,
             url=reverse("notifications:list"),
         )
+
+
+class FileUpdateMixin(ObjectPermissionRequiredMixin, TemplateView):
+    form_class = NewFileUploadForm
+    template_name = "components/object_files_update.html"
+    raise_exception = True
+
+    def get_permission_object(self):
+        return self.base_object
+
+    @cached_property
+    def interface(self):
+        return ComponentInterface.objects.get(
+            slug=self.kwargs["interface_slug"]
+        )
+
+    def get_context_data(self, *args, **kwargs):
+        context = super().get_context_data(*args, **kwargs)
+        context.update(
+            {
+                "form": self.form_class(**self.get_form_kwargs()),
+            }
+        )
+        return context
+
+    def get_form_kwargs(self):
+        return {

```

Comment:
Missing super.

Reply:
Since we're inheriting from a TemplateView super doesn't have the `get_form_kwargs` method. This used to be a FormView, but since we're only using this view to get the correct form, not to post any data to it, I changed it to a TemplateView, just like I did for the InterfacesCreate view in yesterdays PR.

Reply:
Ah, I see, that's a little confusing. Do you need to override `get_form_kwargs` in the subclasses? If not, just throw them in directly.

Reply:
I don't need to override the method in the subclass, so will indeed just add them to the form directly. Thanks!